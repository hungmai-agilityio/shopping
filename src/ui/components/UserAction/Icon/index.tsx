import { Icon } from '@/ui/components';

interface UserIconProps {
  src: string;
  alt: string;
  onClick?: () => void;
}

const UserIcon = ({ src, alt, onClick }: UserIconProps) => {
  return (
    <Icon
      src={src}
      alt={alt}
      width={24}
      height={24}
      onClick={onClick}
      priority
    />
  );
};

export default UserIcon;
