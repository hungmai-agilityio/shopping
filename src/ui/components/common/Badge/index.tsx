import { ReactNode } from 'react';

interface BadgeProps {
  quantity: number;
  children: ReactNode;
}

const Badge = ({ quantity, children }: BadgeProps) => {
  return (
    <div className="relative">
      {quantity > 0 && (
        <div className="absolute -top-2 -right-2 h-5 w-5 rounded-full flex justify-center items-center bg-red-400 text-white text-xs cursor-pointer z-10 overflow-hidden">
          {quantity}
        </div>
      )}
      {children}
    </div>
  );
};

export default Badge;
