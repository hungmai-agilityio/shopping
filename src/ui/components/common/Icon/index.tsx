import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

interface IconProps extends ImageProps {
  onClick?: () => void;
  isDisabled?: boolean;
}

const Icon = ({
  src,
  alt,
  width,
  height,
  onClick,
  isDisabled,
  ...props
}: IconProps) => {
  return (
    <div
      className={clsx('relative inline-flex items-center justify-center', {
        'opacity-20': isDisabled,
        'cursor-pointer': onClick
      })}
      onClick={onClick}
    >
      <Image src={src} alt={alt} width={width} height={height} {...props} />
    </div>
  );
};

export default Icon;
