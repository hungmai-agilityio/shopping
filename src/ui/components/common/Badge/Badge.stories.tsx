import { Meta, StoryObj } from '@storybook/react';

// Components
import Badge from '@/ui/components/common/Badge';
import Icon from '@/ui/components/common/Icon';

const meta: Meta<typeof Badge> = {
  component: Badge,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    quantity: {
      description: 'Display quantity'
    },
    children: {
      description: 'Content or icon show badge'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Badge>;

export const Default: Story = {
  args: {
    quantity: 10,
    children: <Icon src="/cart.svg" alt="cart" width={24} height={24} />
  }
};
