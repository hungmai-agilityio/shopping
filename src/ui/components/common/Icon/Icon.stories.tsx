import { Meta, StoryObj } from '@storybook/react';

// Components
import Icon from '@/ui/components/common/Icon';

const meta: Meta<typeof Icon> = {
  component: Icon,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    src: {
      description: 'Image to render'
    },

    alt: {
      description: 'Describes the image and shows if the image is corrupted'
    },

    width: {
      description: 'Width for images'
    },

    height: {
      description: 'Height for images'
    },

    onClick: {
      description: 'Handle the event when clicking on the button'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Icon>;

export const Default: Story = {
  args: {
    src: '/user.svg',
    alt: 'Cart Icon',
    width: 24,
    height: 24,
    onClick: () => {
      alert(123);
    }
  }
};

export const NotAction: Story = {
  args: {
    src: '/star.svg',
    alt: 'Cart Icon',
    width: 106,
    height: 18
  }
};
