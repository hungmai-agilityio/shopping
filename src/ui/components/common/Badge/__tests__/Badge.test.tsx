import Badge from '@/ui/components/common/Badge';
import Icon from '@/ui/components/common/Icon';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Badge component', () => {
  const onClickMock = jest.fn();

  const defautProps = {
    quantity: 10,
    children: <Icon src="/cart.svg" alt="cart" width={24} height={24} />
  };

  test('renders badge with default props', () => {
    render(<Badge {...defautProps} />);

    const icon = screen.getByAltText('cart');
    expect(icon).toBeInTheDocument();
    const badge = screen.getByText('10');
    expect(badge).toHaveClass(
      'flex justify-center items-center bg-red-400 text-white text-xs cursor-pointer z-10 overflow-hidden'
    );
  });

  test('should be render match to snapshot', () => {
    const { container } = render(<Badge {...defautProps} />);

    expect(container).toMatchSnapshot();
  });
});
