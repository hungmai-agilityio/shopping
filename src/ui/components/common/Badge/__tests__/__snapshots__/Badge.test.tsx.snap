// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should be render match to snapshot 1`] = `
<div>
  <div
    class="relative"
  >
    <div
      class="absolute -top-2 -right-2 h-5 w-5 rounded-full flex justify-center items-center bg-red-400 text-white text-xs cursor-pointer z-10 overflow-hidden"
    >
      10
    </div>
    <div
      class="relative inline-flex items-center justify-center"
    >
      <img
        alt="cart"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/cart.svg"
        style="color: transparent;"
        width="24"
      />
    </div>
  </div>
</div>
`;
